app/controllers/results_controller.rb

```diff
@@ -0,0 +1,29 @@
+class ResultsController < ApplicationController
+  def index
+  end
+
+  def show
+    @path = params[:base_path]
+    @path = URI.parse(@path).path if @path.starts_with?('http')
```

Comment:
Some documents in the index actually start their link field with "http"; eg, http://www.nhs.uk/

I think this parsing should either be removed, or only apply if the path starts with https://www.gov.uk